Mergin/utils.py

```diff
@@ -602,59 +607,112 @@ def package_layer(layer, project_dir):
+    If it is a GeoPackage raster, save it also as a GeoPackage table.
+    Otherwise, save the raster as GeoTiff using Qgs with some optimisations.
+    """
 
-        provider_opts = QgsDataProvider.ProviderOptions()
-        provider_opts.layerName = layer.name()
-        datasource = layer_filename
-        layer.setDataSource(datasource, layer.name(), "gdal", provider_opts)
+    driver_name = get_raster_driver_name(raster_layer)
 
+    if driver_name == "GTiff":
+        copy_tif_raster(raster_layer, project_dir)
+    elif driver_name == "GPKG":
+        save_raster_to_geopackage(raster_layer, project_dir)
     else:
-        # meshes and anything else
-        return False
-    return True
+        save_raster_as_geotif(raster_layer, project_dir)
+
+
+def copy_tif_raster(raster_layer, project_dir):
+    """Create a copy of raster layer file(s) without rewriting."""
+    dp = raster_layer.dataProvider()
+    src_filepath = dp.dataSourceUri() if os.path.isfile(dp.dataSourceUri()) else None
+    if src_filepath is None:
+        raise PackagingError(f"Can't find the source file for {raster_layer.name()}")
+
+    # Make sure the destination path is unique and create a copy including any external pyramids
+    new_raster_filename = get_unique_filename(os.path.join(project_dir, os.path.basename(src_filepath)))
+    shutil.copy(src_filepath, new_raster_filename)
+    # External pyramids
+    if os.path.exists(src_filepath + ".ovr"):
+        shutil.copy(src_filepath + ".ovr", new_raster_filename + ".ovr")
+    # Erdas external pyramids
+    src_filepath_no_ext = os.path.splitext(src_filepath)[0]
+    if os.path.exists(src_filepath_no_ext + ".aux"):
+        shutil.copy(src_filepath_no_ext + ".aux", os.path.splitext(new_raster_filename)[0] + ".aux")
+
+
+def save_raster_to_geopackage(raster_layer, project_dir):
+    """Save a GeoPackage raster to GeoPackage table in the project directory."""
+    layer_filename = get_unique_filename(os.path.join(project_dir, raster_layer.name() + ".gpkg"))
+
+    raster_writer = QgsRasterFileWriter(layer_filename)
+    raster_writer.setOutputFormat("gpkg")
+    raster_writer.setCreateOptions([f"RASTER_TABLE={raster_layer.name()}", "APPEND_SUBDATASET=YES"])
+
+    write_raster(raster_layer, raster_writer, layer_filename)
+
+
+def save_raster_as_geotif(raster_layer, project_dir):
+    """Save raster layer as GeoTiff in the project directory."""
+    new_raster_filename = get_unique_filename(raster_layer.name() + ".tif")
+    layer_filename = os.path.join(project_dir, os.path.basename(new_raster_filename))
+    # Get data type info to set a proper compression type
+    dp = raster_layer.dataProvider()
+    is_byte_data = [dp.dataType(i) <= Qgis.Byte for i in range(raster_layer.bandCount())]
+    compression = "JPEG" if all(is_byte_data) else "LZW"
+
+    raster_writer = QgsRasterFileWriter(layer_filename)
+    raster_writer.setCreateOptions([f"COMPRESS={compression}"])
```

Comment:
I would suggest to also use `TILED=YES` - otherwise pixels are stored line by line which is typically less efficient (especially for larger rasters)